test/pkg.jl

```diff
@@ -141,6 +141,27 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+
+    #-------
+    # 17364
+    #-------
+    let branch_name = "test-branch",
+        branch_commit = "ba3888212e30a7974ac6803a89e64c7098f4865e"
+
+        # create a branch in Example package
+        LibGit2.with(LibGit2.GitRepo, Pkg.dir("Example")) do repo
+            LibGit2.branch!(repo, branch_name, branch_commit, set_head=false)
```

Comment:
@wildart bump, can we please get this test cleaned up so it's actually representative of the failure and get this fix into rc1?